sqrl-calcite/src/main/java/com/datasqrl/plan/validate/ScriptPlanner.java

```diff
@@ -221,40 +229,64 @@ public Void visit(SqrlImportDefinition node, Void context) {
   public Void visit(SqrlExportDefinition node, Void context) {
     NamePath sinkPath = nameUtil.toNamePath(node.getSinkPath().names);
 
-    // Resolve through connector factory or module loader
-    Optional<TableSink> sink = connectorFactoryFactory.create(null, sinkPath.getFirst().getCanonical())
-        .map(t->t.createSourceAndSink(new ConnectorFactoryContext(sinkPath.getLast(),
-            Map.of("name", sinkPath.getLast().getDisplay()))))
-        .map(t->TableSinkImpl.create(t, sinkPath, Optional.empty()))
-        .or(()->moduleLoader.getModule(sinkPath.popLast())
-            .flatMap(m -> m.getNamespaceObject(sinkPath.getLast()))
-            .filter(t->t instanceof TableSinkObject)
-            .map(t->((TableSinkObject) t).getSink()));
-
-    if (sink.isEmpty()) {
-      addError(ErrorCode.CANNOT_RESOLVE_TABLESINK, node, "Cannot resolve table sink: %s",
-          nameUtil.toNamePath(node.getSinkPath().names).getDisplay());
-      return null;
-    }
-
     NamePath path = nameUtil.toNamePath(node.getIdentifier().names);
 
     QueryPlanner planner = framework.getQueryPlanner();
     Optional<RelOptTable> table = planner.getCatalogReader().getTableFromPath(path);
     Preconditions.checkState(table.isPresent(), "Could not find export table: %s", path.getDisplay());
-    ResolvedExport resolvedExport = exportTable(table.get().unwrap(ModifiableTable.class), sink.get(), planner.getRelBuilder(), true);
+    ModifiableTable modTable = table.get().unwrap(ModifiableTable.class);
+
+    int numSelects = modTable.getNumSelects();
+    RelNode exportRelNode = planner.getRelBuilder().scan(modTable.getNameId()).build();
+
+    Optional<SystemBuiltInConnectors> builtInSink = SystemBuiltInConnectors.forExport(sinkPath.popLast());
+    TableSink tableSink = null;
+    if (builtInSink.isPresent()) {
+      SystemBuiltInConnectors connector = builtInSink.get();
+      if (connector == SystemBuiltInConnectors.LOG) {
```

Comment:
this entire if block doesn't seem necessary. They are provided by the connector factory. The connector factory could bookkeep easily since it is aware of the type of sink its creating

Reply:
Do you mean bookkeeping the logs? This part is about using the LogManager to bookkeep the logs which is probably more suited for that task than a factory.